components/bl-particles-component/src/helpers.js

```diff
@@ -0,0 +1,191 @@
+    },
+    roll      : { enable: rollingAnimation, speed },
+    color     : {
+      value    : color || Colors.RANDOM,
+      animation: { enable: colorAnimation, speed: COLOR_ANIMATION_SPEED },
+    },
+    opacity   : { value: opacity },
+    shape     : {
+      type   : shape,
+      options: {
+        [Shapes.IMAGE] : Array.isArray(imageURL) ? imageURL : { src: imageURL },
+        [Shapes.TEXT]  : { value: textValue },
+        [Shapes.SPIRAL]: SpiralShapeProps,
+        [Shapes.LINE]  : LineShapeProps,
+      },
+    },
+    move      : { enable: move, direction, speed, outModes },
+    size      : { value: size },
+    number    : { value: number, density: { enable: true } },
+  }), [
+    collisionMode, lineLinksVisibility, triangleLinksVisibility, linksColor, rotate, rotationAnimation, speed, number,
+    rollingAnimation, color, colorAnimation, opacity, shape, imageURL, textValue, move, direction, outModes, size,
+  ]);
+
+  return useMemo(() => ({
+    autoPlay, particles,
+    duration    : duration || DEFAULT_TIMING,
+    delay       : delay || DEFAULT_TIMING,
+    fullScreen  : { enable: fullScreen, zIndex: zIndex || DEFAULT_Z_INDEX },
+    background  : { color: backgroundColor, opacity: backgroundOpacity },
+    detectRetina: true,
+    pauseOnBlur : false,
+    ...PresetsMap[preset],
+    ...customOptions,
+  }), [
+    autoPlay, particles, duration, delay, fullScreen, zIndex, backgroundColor, backgroundOpacity, preset, customOptions,
+  ]);
+}
+
+async function loadShapes(engine) {
+  await loadBubbleShape(engine);
+  await loadCardsShape(engine);
+  await loadHeartShape(engine);
+  await loadMultilineTextShape(engine);
+  await loadPathShape(engine);
+  await loadRoundedRectShape(engine);
+  await loadSpiralShape(engine);
+}
+
+async function loadPlugins(engine) {
+  await loadHsvColorPlugin();
+  await loadEasingBackPlugin();
+  await loadEasingCircPlugin();
+  await loadEasingCubicPlugin();
+  await loadEasingExpoPlugin();
+  await loadEasingQuadPlugin();
+  await loadEasingQuartPlugin();
+  await loadEasingQuintPlugin();
+  await loadEasingSinePlugin();
+  await loadPolygonMaskPlugin(engine);
+  await loadCanvasMaskPlugin(engine);
+  await loadMotionPlugin(engine);
+  await loadSoundsPlugin(engine);
+  await loadInfectionPlugin(engine);
+}
+
+async function loadUpdaters(engine) {
+  await loadGradientUpdater(engine);
+  await loadOrbitUpdater(engine);
+}
+
+async function loadPaths(engine) {
+  await loadCurvesPath(engine);
+  await loadPerlinNoisePath(engine);
+  await loadPolygonPath(engine);
+  await loadSimplexNoisePath(engine);
+}
+
+async function loadInteractions(engine) {
+  await loadLightInteraction(engine);
+  await loadParticlesRepulseInteraction(engine);
+}
+
+export async function loadParticles(engine) {
+  await loadFull(engine);
+  await loadPlugins(engine);
+  await loadInteractions(engine);
+  await loadUpdaters(engine);
+  await loadPaths(engine);
+  await loadShapes(engine);
+}
```

Comment:
Do we need to do it synchronously?
`Promise.all` if no.

Also, all of this functions needed for every particles type?

Reply:
Not all of these functions are required for each particle type, but `onInit` with `loadParticles` runs only once on the first render, so we have to consider all possible configurations.